### PR TITLE
Move renderTexture pool from FilterSystem

### PR DESCRIPTION
**Pull request by ivanpopelyshev**
Some filters can require custom pools, made from texture of different format.

It also allows for a plugin to override FilterSystem texturePool behaviour.

Who knows, maybe soon we'll need a pool of multi-sample framebuffers or sRGB stuff!

### CODE REVIEW

**Comment by ivanpopelyshev** [comment]
Custom pool usage for FLOAT textures: https://www.pixiplayground.com/#/edit/4rskiJ8lb8k6437WmSwKy 

Need demo for filter-heavy scene to check for other stupid errors

File: packages/core/src/renderTexture/RenderTexturePool.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
+import RenderTexture from './RenderTexture';
+import BaseRenderTexture from './BaseRenderTexture';
+import { nextPow2 } from '@pixi/utils';
+
+/**
+ * Experimental!
+ *
+ * Texture pool, used by FilterSystem and plugins
+ * Stores collection of temporary pow2 or screen-sized renderTextures
```
Comment (bigtimebuddy) [inline packages/core/src/renderTexture/RenderTexturePool.js]: Do you want to add a doc `@class` etc for this?

File: packages/core/src/renderTexture/RenderTexturePool.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    returnFilterTexture(renderTexture)
+    {
+        this.returnTexture(renderTexture);
+    }
+
+    /**
+     * Clears the pool
+     *
+     * @member {boolean} [destroyTextures=true] destroy all stored textures
+     */
```
Comment (GoodBoyDigital) [inline packages/core/src/renderTexture/RenderTexturePool.js]: could this be:
`clear(destroyTextures = false)`

File: packages/core/src/renderTexture/RenderTexturePool.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                textures[j].destroy(true);
+            }
+        }
+        this.texturePool[screenKey] = [];
+
+        this._pixelsWidth = this.renderer.view.width;
+        this._pixelsHeight = this.renderer.view.height;
+    }
+}
+
```
Comment (GoodBoyDigital) [inline packages/core/src/renderTexture/RenderTexturePool.js]: what is Screen Key used for? Mind adding some docs for it?

File: packages/core/src/renderTexture/RenderTexturePool.js (hunk 2 above)
Comment (ivanpopelyshev) [reply packages/core/src/renderTexture/RenderTexturePool.js]: no, its true by default, like in previous version.

File: packages/core/src/renderTexture/RenderTexturePool.js (hunk 3 above)
Comment (ivanpopelyshev) [reply packages/core/src/renderTexture/RenderTexturePool.js]: ok

File: packages/core/src/renderTexture/RenderTexturePool.js (hunk 2 above)
Comment (ivanpopelyshev) [reply packages/core/src/renderTexture/RenderTexturePool.js]: i dont know how to do that part, and its internal thing for a custom class, we can change it later.

**Comment by ivanpopelyshev** [comment]
I've so added `ISize` type, that way we don't need to pass renderer to renderTexturePool.

**Comment by ivanpopelyshev** [comment]
https://www.pixiplayground.com/#/edit/R9DigjpNQlmI~vrs~4j~T

output is 1, 0, 1

**Pull request merged**